src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.CacheEntryTokens.cs

```diff
@@ -29,9 +29,8 @@ internal void AttachTokens(CacheEntry cacheEntry)
                 {
                     lock (this)
                     {
-                        for (int i = 0; i < _expirationTokens.Count; i++)
+                        foreach (IChangeToken expirationToken in _expirationTokens)
```

Comment:
For loop should be much more efficient, no? (eg. no enumerator to dispose)

cc: @stephentoub

Reply:
For loop over a `List<T>` is likely slightly more efficient than foreach over it, yes.

Reply:
Yes, I `List<>` enumeration has some overhead. Changed back to for-loop, but using a local variable to cache the list value.